src/Media.Plugin.Android/MediaImplementation.cs

```diff
@@ -604,45 +622,57 @@ public Task<bool> ResizeAsync(string filePath, PhotoSize photoSize, int quality,
             }
         }
 
-
-        void SetExifData(string filePath, Orientation orientation)
+        void SetMissingMetadata(ExifInterface exif, Location location)
         {
-            try
+            Single[] position = new Single[6];
+            if (!exif.GetLatLong(position) && location.Latitude+location.Longitude > 0)
             {
-                using (var ei = new ExifInterface(filePath))
-                {
-
-                    ei.SetAttribute(ExifInterface.TagOrientation, Java.Lang.Integer.ToString((int)orientation));
-                    ei.SaveAttributes();                    
-                }
+                exif.SetAttribute(ExifInterface.TagGpsLatitude, coordsToDMS(location.Latitude));
+                exif.SetAttribute(ExifInterface.TagGpsLongitude, coordsToDMS(location.Longitude));
+                exif.SetAttribute(ExifInterface.TagGpsLatitudeRef, location.Latitude > 0 ? "N" : "S");
+                exif.SetAttribute(ExifInterface.TagGpsLongitudeRef, location.Longitude > 0 ? "E" : "W");
             }
-            catch (Exception ex)
+            if (string.IsNullOrEmpty(exif.GetAttribute(ExifInterface.TagDatetime)))
             {
-#if DEBUG
-                throw ex;
-#endif
+                exif.SetAttribute(ExifInterface.TagDatetime, DateTime.Now.ToString("yyyy:MM:dd hh:mm:ss"));
+            }
+            if (string.IsNullOrEmpty(exif.GetAttribute(ExifInterface.TagMake))) {
+                exif.SetAttribute(ExifInterface.TagMake, Build.Manufacturer);
+            }
+            if (string.IsNullOrEmpty(exif.GetAttribute(ExifInterface.TagModel)))
+            {
+                exif.SetAttribute(ExifInterface.TagModel, Build.Model);
             }
         }
 
-        static int GetRotation(string filePath)
+        private string coordsToDMS(double coord)
```

Comment:
This should be ```CoordinatesTo... whatever DMS is...``` should be more descritive of a title, not sure what it is doing.